cdragontoolbox/wad.py

```diff
@@ -28,6 +29,13 @@ def test_jpeg_photoshop(h, f):
 # read file data, caching will reduce I/Os
 _hash_to_guessed_extensions = {}
 
+class MalformedSubchunkException(BaseException):
+    """Subchunk data is invalid or doesn't match the provided subchunktoc"""
+
+    wad_data = None
```

Comment:
Why having this attribute on the class itself?

Reply:
I was going by your suggestion of having the invalid data stored in the exception. I don't actually know how exactly exceptions work so I don't know if there is a better way to do this.

Reply:
I'm not an expert on the good practices for custom exceptions with data (other than messages), but it seems fine.
But you don't have to declare `wad_data` here. Using `self.wad_data` in the constructor is enough.